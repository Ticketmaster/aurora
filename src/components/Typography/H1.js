import styled from "styled-components";

import { fontWeight, fontSize, BASE_STYLE_HEADER } from "../../mixins";
import { largeAndUp } from "../../theme/mediaQueries";
import { EXTRA_BOLD, ONYX, BASE } from "../constants";
import typographyClassnames from "./helpers";

const H1 = styled.h1.attrs(
  ({ weight = EXTRA_BOLD, color = ONYX, variant = BASE }) => ({
    className: typographyClassnames("h1", { weight, color, variant })
  })
)`
  ${fontWeight`extraBold`}
  ${BASE_STYLE_HEADER}
  ${fontSize`zetta`} ${largeAndUp`${fontSize`bronto`}`};
`;

H1.displayName = "Typography.H1";

export default H1;
