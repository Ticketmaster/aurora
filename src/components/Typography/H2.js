import styled from "styled-components";

import { fontWeight, fontSize, BASE_STYLE_HEADER } from "../../mixins";
import { largeAndUp } from "../../theme/mediaQueries";
import { REGULAR, ONYX, BASE } from "../constants";
import typographyClassnames from "./helpers";

const H2 = styled.h2.attrs(
  ({ weight = REGULAR, color = ONYX, variant = BASE }) => ({
    className: typographyClassnames("h2", { weight, color, variant })
  })
)`
  ${fontWeight`regular`}
  ${BASE_STYLE_HEADER}
  ${fontSize`tera`} ${largeAndUp`${fontSize`zetta`}`};
`;

H2.displayName = "H2";

export default H2;
