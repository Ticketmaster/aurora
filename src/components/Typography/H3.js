import styled from "styled-components";

import { fontWeight, fontSize, BASE_STYLE_HEADER } from "../../mixins";
import { largeAndUp } from "../../theme/mediaQueries";
import { REGULAR, ONYX, BASE } from "../constants";
import typographyClassnames from "./helpers";

const H3 = styled.h3.attrs(
  ({ weight = REGULAR, color = ONYX, variant = BASE }) => ({
    className: typographyClassnames("h3", { weight, color, variant })
  })
)`
  ${fontWeight`regular`}
  ${BASE_STYLE_HEADER}
  ${fontSize`giga`} ${largeAndUp`${fontSize`tera`}`};
`;

H3.displayName = "Typography.H3";

export default H3;
