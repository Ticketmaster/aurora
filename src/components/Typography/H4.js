import styled from "styled-components";

import { fontWeight, fontSize, BASE_STYLE_HEADER } from "../../mixins";
import { REGULAR, ONYX, BASE } from "../constants";
import typographyClassnames from "./helpers";

const H4 = styled.h4.attrs(
  ({ weight = REGULAR, color = ONYX, variant = BASE }) => ({
    className: typographyClassnames("h4", { weight, color, variant })
  })
)`
  ${fontWeight`regular`}
  ${BASE_STYLE_HEADER}
  ${fontSize`giga`};
`;

H4.displayName = "Typography.H4";

export default H4;
