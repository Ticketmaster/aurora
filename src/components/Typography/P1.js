import styled from "styled-components";

import { fontSize, BASE_STYLE_BODY } from "../../mixins";
import { REGULAR, ONYX, BASE } from "../constants";
import typographyClassnames from "./helpers";

const P1 = styled.p.attrs(
  ({ weight = REGULAR, color = ONYX, variant = BASE }) => ({
    className: typographyClassnames("p1", { weight, color, variant })
  })
)`
  ${BASE_STYLE_BODY};
  ${fontSize`kilo`};
`;

P1.displayName = "Typography.P1";

export default P1;
