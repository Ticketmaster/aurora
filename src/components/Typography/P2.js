import styled from "styled-components";
import classnames from "classnames";

import { fontSize, BASE_STYLE_BODY } from "../../mixins";
import { REGULAR, ONYX, BASE } from "../constants";

const P2 = styled.p.attrs(
  ({ weight = REGULAR, color = ONYX, variant = BASE }) => ({
    className: classnames({
      p2: true,
      [`p2--${weight}`]: !!weight,
      [`p2--${color}-${variant}`]: !!(color && variant)
    })
  })
)`
  ${BASE_STYLE_BODY};
  ${fontSize`hecto`};
`;

export default P2;
