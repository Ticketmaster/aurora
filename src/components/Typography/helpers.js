import classnames from "classnames";

const typographyClassnames = (
  elem,
  { weight, color, variant, textTransform } = {}
) =>
  elem
    ? classnames({
        [elem]: true,
        [`${elem}--${weight}`]: !!weight,
        [`${elem}--${textTransform}`]: !!textTransform,
        [`${elem}--${color}-${variant}`]: !!(color && variant)
      })
    : "";

export default typographyClassnames;
