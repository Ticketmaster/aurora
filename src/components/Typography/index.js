import H1 from "./H1";
import H2 from "./H2";
import H3 from "./H3";
import H4 from "./H4";
import P1 from "./P1";
import P2 from "./P2";
import P3 from "./P3";
import P4 from "./P4";

// sets a default & fallback
const Typography = P1;

Typography.H1 = H1;
Typography.H2 = H2;
Typography.H3 = H3;
Typography.H4 = H4;
Typography.P1 = P1;
Typography.P2 = P2;
Typography.P3 = P3;
Typography.P4 = P4;

export default Typography;
