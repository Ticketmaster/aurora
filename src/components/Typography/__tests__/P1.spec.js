import React from "react";
import renderer from "react-test-renderer";

import Typography from "../";
import MOCK_STRING from "../__mocks__/string.mocks";

describe("Typography.P1", () => {
  it("should render default markup when no props are passed", () => {
    expect(
      renderer.create(<Typography.P1>{MOCK_STRING}</Typography.P1>)
    ).toMatchSnapshot();
  });

  it("should render with a custom weight prop", () => {
    expect(
      renderer.create(
        <Typography.P1 weight="semiBold">{MOCK_STRING}</Typography.P1>
      )
    ).toMatchSnapshot();
  });

  it("should render with custom color and variant props", () => {
    expect(
      renderer.create(
        <Typography.P1 color="accent01" variant="dark">
          {MOCK_STRING}
        </Typography.P1>
      )
    ).toMatchSnapshot();
  });

  it("should render with a custom color prop that lacks variants", () => {
    expect(
      renderer.create(
        <Typography.P1 color="brand" variant="">
          {MOCK_STRING}
        </Typography.P1>
      )
    ).toMatchSnapshot();
  });

  it("should render without children", () => {
    expect(renderer.create(<Typography.P1 />)).toMatchSnapshot();
  });
});
