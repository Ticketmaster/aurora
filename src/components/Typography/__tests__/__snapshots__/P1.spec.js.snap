// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography.P1 should render default markup when no props are passed 1`] = `
<p
  className="p1 p1--regular p1--onyx-base grArvb"
>
  Hello world
</p>
`;

exports[`Typography.P1 should render with a custom color prop that lacks variants 1`] = `
<p
  className="p1 p1--regular p1--brand-base hknJbF"
  color="brand"
>
  Hello world
</p>
`;

exports[`Typography.P1 should render with a custom weight prop 1`] = `
<p
  className="p1 p1--semiBold p1--onyx-base iQqweo"
>
  Hello world
</p>
`;

exports[`Typography.P1 should render with a textTransform prop 1`] = `
<p
  className="p1 p1--regular p1--onyx-base dImeXY"
>
  Hello world
</p>
`;

exports[`Typography.P1 should render with custom color and variant props 1`] = `
<p
  className="p1 p1--regular p1--accent01-dark cwYgwV"
  color="accent01"
>
  Hello world
</p>
`;

exports[`Typography.P1 should render without children 1`] = `
<p
  className="p1 p1--regular p1--onyx-base grArvb"
/>
`;
